flow/scenarios/base_scenario.py

```diff
@@ -548,14 +548,80 @@ def specify_connections(self, net_params):
     def specify_routes(self, net_params):
         """Specify the routes vehicles can take starting from any edge.
 
-        The routes are specified as lists of edges the vehicle must traverse,
-        with the first edge corresponding to the edge the vehicle begins on.
-        Note that the edges must be connected for the route to be valid. If
-        this method is not implemented, vehicles that enter a network are
-        assigned routes consisting solely on their current edges, and exit the
-        network once they reach the end of their edge.
-
-        Currently, only one route is allowed from any given starting edge.
+        Routes can be specified in one of three ways:
+
+        * In this case of deterministic routes (as is the case in the ring road
+          scenario), the routes can be specified as dictionary where the key
+          element represents the starting edge and the element is a single list
+          of edges the vehicle must traverse, with the first edge corresponding
+          to the edge the vehicle begins on. Note that the edges must be
+          connected for the route to be valid.
+
+          For example (from flow/scenarios/loop.py):
+
+          >>> def specify_routes(self, net_params):
+          >>>     return {
+          >>>         "top": ["top", "left", "bottom", "right"],
+          >>>         "left": ["left", "bottom", "right", "top"],
+          >>>         "bottom": ["bottom", "right", "top", "left"],
+          >>>         "right": ["right", "top", "left", "bottom"]
+          >>>     }
+
+        * Alternatively, if the routes are meant to be stochastic, each element
+          can consist of a list of (route, probability) tuples, where the first
+          element in the tuple is one of the routes a vehicle can take from a
+          specific starting edge, and the second element is the probability
+          that vehicles will choose that route. Note that, in this case, the
+          sum of probability values for each dictionary key must sum up to one.
+
+          For example, modifying the code snippet we presented above, another
+          valid way of representing the route in a more probabilistic setting
+          is:
+
+          >>> def specify_routes(self, net_params):
```

Comment:
Can you give a more representative example of this case? Perhaps something like:
```
"top-random": [
    (["top", "left", "bottom", "right"], 0.9)
    (["top", "right", "bottom", "left"], 0.1)
    ],
```

Reply:
done